calcrule_social_protection/strategies/benefit_package_base_strategy.py

```diff
@@ -22,13 +23,25 @@ def calculate(cls, calculation, payment_plan, **kwargs):
         audit_user_id, start_date, end_date = \
             calculation.get_payment_cycle_parameters(**kwargs)
         user = User.objects.filter(i_user__id=audit_user_id).first()
+
+        #  :TODO ticket - at this stage use fixed amount, also skip ceiling part
+        payment = payment_plan_parameters['calculation_rule']['fixed_batch']
+        advanced_filters_criteria = payment_plan_parameters['advanced_criteria']
         for beneficiary in beneficiares:
-            # TODO add calculation mechanism once is developed as a part of CM-210
-            #  ticket - at this stage use fixed amount
-            fixed_amount = payment_plan_parameters['calculation_rule']['fixed_batch']
+            for criterion in advanced_filters_criteria:
+                condition = criterion['custom_filter_condition']
+                calculated_amount = float(criterion['amount'])
+                does_amount_apply_for_limitations = criterion['count_to_max']
+                if cls._does_beneficiary_meet_condition(beneficiary, condition):
+                    if does_amount_apply_for_limitations:
+                        # TODO: ceiling part
+                        continue
```

Comment:
'pass' is also a good choice -  it should behave in the same way (also do nothing, and we can go to next iteration). I usually use 'pass' rather than 'continue'.  @jdolkowski